### ISSUE TITLE
Error: failed to start faucet: Unable to bind faucet to 0.0.0.0:9900

### ISSUE THREAD

**Issue by SwayStar123** [Ackee-Blockchain/trdelnik#68]
When I first try to run trdelnik test, it fails saying localnet is not running, and then if i retry, it says the port is already used, so im thinking its not waiting long enough or something?
```

-------------- TEST START --------------
Error: localnet is not running
thread 'test_happy_path' panicked at 'assertion failed: `(left == right)`
  left: `1`,
 right: `0`: the test returned a termination value with a non-zero status code (1) which indicates a failure', /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/test/src/lib.rs:185:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
test test_happy_path ... FAILED

failures:

failures:
    test_happy_path

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 5.10s

error: test failed, to rerun pass '-p trdelnik-tests --test test'
Error: testing failed
swaystar@Babaji:~/programming/solana/altruism$ Waiting for fees to stabilize 1...

swaystar@Babaji:~/programming/solana/altruism$ trdelnik test
warning: profiles for the non root package will be ignored, specify profiles at the workspace root:
package:   /home/swaystar/programming/solana/altruism/programs/altruism/Cargo.toml
workspace: /home/swaystar/programming/solana/altruism/Cargo.toml
warning: unused import: `altruism`
 --> trdelnik-tests/tests/test.rs:5:5
  |
5 | use altruism;
  |     ^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

warning: `trdelnik-tests` (test "test") generated 1 warning
    Finished test [unoptimized + debuginfo] target(s) in 0.40s
     Running unittests src/lib.rs (target/debug/deps/altruism-4cce7156c9d72dbd)

running 1 test
test test_id ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running unittests src/lib.rs (target/debug/deps/program_client-316725de8b77c5f3)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running tests/test.rs (target/debug/deps/test-429cbf0c0b1f2f8b)

running 1 test
-------------- TEST START --------------
Error: failed to start faucet: Unable to bind faucet to 0.0.0.0:9900, check the address is not already in use: Address already in use (os error 98)
program_keypair(1).pubkey = Po1RaS8BEDbNcn5oXsFryAeQ6Wn8fvmE111DJaKCgPC
program_keypair(1).pubkey = Po1RaS8BEDbNcn5oXsFryAeQ6Wn8fvmE111DJaKCgPC
Error: localnet is still running
thread 'test_happy_path' panicked at 'assertion failed: `(left == right)`
  left: `1`,
 right: `0`: the test returned a termination value with a non-zero status code (1) which indicates a failure', /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/test/src/lib.rs:185:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
test test_happy_path ... FAILED

failures:

failures:
    test_happy_path

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 4.39s

error: test failed, to rerun pass '-p trdelnik-tests --test test'
Error: testing failed
```

**Comment by SwayStar123** [comment]
https://github.com/SwayStar123/altruism

Here is the entire anchor project

**Comment by stefanprokopdev** [comment]
Hey @SwayStar123,

thanks for your issue! Can you please tell me which version of anchor CLI do you use locally? In your code you work with anchor 0.24 which should be ok for trdelnik but what version of CLI do you use?

Trdelnik should be optimized for anchor < 0.25 (need to fix it in our readme). We currently exploring the version 0.25 so it's not supported yet.

**Comment by SwayStar123** [comment]
@stefanprokopdev i tried both 0.25 and 0.24 currently using 0.24

**Comment by antoninkriz** [comment]
@SwayStar123 Hi! We've just released a new version `0.2.0` with configurable startup timeout for `solana-test-validator`.
Make sure to check [the `CHANGELOG.md`](https://github.com/Ackee-Blockchain/trdelnik/blob/4b7b20cdf3a729442f958b600c46df31ca1a13be/CHANGELOG.md) and the `Configuration` section in [the `README.md`](https://github.com/Ackee-Blockchain/trdelnik/blob/4b7b20cdf3a729442f958b600c46df31ca1a13be/README.md#configuration) as this is a breaking change.

**Comment by antoninkriz** [comment]
@SwayStar123 I'm closing this issue as I consider it resolved based on our conversation in #71
